python/plugins/processing/core/ProcessingConfig.py

```diff
@@ -83,6 +84,10 @@ def initialize():
             ProcessingConfig.tr('Use only selected features'), True))
         ProcessingConfig.addSetting(Setting(
             ProcessingConfig.tr('General'),
+            ProcessingConfig.IGNORE_NULL_GEOMS,
+            ProcessingConfig.tr('Ignore features with null geometries'), True))
```

Comment:
I think algorithms shouldn't decide how to deal with null geometries. A null geometry is "unknown" not necessarily "non-existent", so basically often they should be conserved as null geometry in the dataset and not silently discarded. On the other hand, this may also be surprising to the user. So it should also not happen silently.

What I would propose is a [fail-fast](https://en.wikipedia.org/wiki/Fail-fast) method: stop the process if null geometries are encountered, explain to the user that he should

> please toggle the (default off) option "include null geometries" or run the "remove null geometries" on layer X to clean null geomtries.

This way, a user is made conscious of possible problems and can deal with them in an appropriate way.